packages/discord-bot/src/AppDiscord.ts

```diff
@@ -1,9 +1,67 @@
-import { ArgsOf, Client, Discord, On } from '@typeit/discord';
+import { Command, CommandMessage, Discord } from '@typeit/discord';
+import { sourcecred as sc } from 'sourcecred-publish-test';
 
-@Discord()
+import { CONFIG } from './config';
+
+export const storage = new sc.ledger.storage.GithubStorage(
+  CONFIG.githubApiToken,
+  'MetaFam/XP',
+);
+
+const manager = new sc.ledger.manager.LedgerManager({
+  storage,
+});
+
+const ledgerLoadedPromise = manager.reloadLedger();
+
+const addressUtils = sc.plugins.ethereum.utils.address;
+
+@Discord('!')
 export abstract class AppDiscord {
-  @On('message')
-  async onMessage([message]: ArgsOf<'message'>, _: Client) {
-    console.log({ message });
+  @Command('setAddress :ethAddress')
+  // @ts-expect-error should be allowed
+  private async setAddress(message: CommandMessage) {
+    const res = await ledgerLoadedPromise;
+    if (res.error) {
+      await message.reply(`Error Loading Ledger: ${res.error}`);
```

Comment:
Had it been me, I would `throw` on an exception and wrap all the logic in a `try`/`catch` rather than `reply`ing & `return`ing in multiple places.